### PR TITLE
TGP-1094 - Audit event start update goods record

### PR DESCRIPTION
**Pull request by yohan-oc**

### CODE REVIEW

File: app/controllers/CategoryGuidanceController.scala [hunk 1]
```diff
@@ -53,14 +55,22 @@ class CategoryGuidanceController @Inject() (
...
           Future.successful(Redirect(routes.JourneyRecoveryController.onPageLoad().url))
       }
   }
 
   // TODO replace index route
-  def onSubmit: Action[AnyContent] = (identify andThen getData) { implicit request =>
-    Redirect(routes.IndexController.onPageLoad.url)
+  def onSubmit(recordId: String): Action[AnyContent] = (identify andThen getData andThen requireData).async {
+    implicit request =>
+      auditService
```
Comment (liandra-giacomine) [inline app/controllers/CategoryGuidanceController.scala:67]: should be a fire and forget so don't map it just execute

File: app/controllers/CategoryGuidanceController.scala [hunk 2]
```diff
@@ -53,14 +55,22 @@ class CategoryGuidanceController @Inject() (
...
+        .auditStartUpdateGoodsRecord(
+          request.eori,
+          request.affinityGroup,
+          "categorisation",
```
Comment (liandra-giacomine) [inline app/controllers/CategoryGuidanceController.scala:71]: can we add this to a constants file?

**Review by liandra-giacomine — COMMENTED** [review]
looks good, just a small comment

File: app/controllers/CategoryGuidanceController.scala (hunk 1 above)
Comment (yohan-oc) [reply app/controllers/CategoryGuidanceController.scala:67]: Please check now :)

**Review by ChrisCookOC — APPROVED** [review]

**Review by liandra-giacomine — APPROVED** [review]

**Review by besscerule — APPROVED** [review]

**Pull request merged**